File: array/1752_check_if_array_is_sorted_and_rotated/1752_check_if_array_is_sorted_and_rotated.py
from typing import List

class Solution:
    def check(self, nums: List[int]) -> bool:
        B = sorted(nums)
        for i in range(len(nums)):
            if nums[i:]+nums[:i] == B:
                return True
        else:
            return False


nums = [3,4,5,1,2]
res = Solution().check(nums)
print(res)

File: array/1752_check_if_array_is_sorted_and_rotated/1752_check_if_array_is_sorted_and_rotated.go
package main

import "fmt"

func check(nums []int) bool {
	cnt := 0
	for i := 0; i < len(nums)-1; i++ {
		if nums[i] > nums[i+1] {
			cnt++
			if cnt > 1 || nums[0] < nums[len(nums)-1] {
				return false
			}
		}

	}
	return true
}

func main() {
	nums := []int{3, 4, 5, 1, 2}
	res := check(nums)
	fmt.Println(res)
}
